src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -166,18 +167,33 @@ def __init__(self, session: "snowflake.snowpark.session.Session") -> None:
         self.generated_alias_maps = {}
         self.subquery_plans = []
         self.alias_maps_to_use: Optional[Dict[uuid.UUID, str]] = None
+        self.config_context: ConfigContext = ConfigContext(session)
+        # Point this plan builder to the same config context as the analyzer
+        self.plan_builder.config_context = self.config_context
 
     def analyze(
         self,
         expr: Union[Expression, NamedExpression],
         df_aliased_col_name_to_real_col_name: DefaultDict[str, Dict[str, str]],
         parse_local_name=False,
+    ) -> str:
+        # Set the config context for analysis step
+        with self.config_context:
```

Comment:
I see analyze is also called during resolve stage, and we create one snapshot when WithStatement is used, does that mean the config snapshot can change during the resolve stage?

Reply:
I updated call from `analyze` -> `do_analyze` in `resolve`. I created a snapshot in `resolve` using with statement. Since a single analyzer shared the same `config_context`, `do_analyze` uses the same context during resolve. same goes for `plan_builder`